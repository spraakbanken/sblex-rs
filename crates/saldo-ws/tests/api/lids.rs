use crate::conftest::spawn_app;
use reqwest::StatusCode;
use rstest::rstest;

#[rstest]
#[case("json")]
#[case("xml")]
#[case("html")]
#[tokio::test]
async fn invalid_input_returns_400(#[case] format: &str) -> eyre::Result<()> {
    // Arrange
    let ctx = spawn_app().await?;
    let client = reqwest::Client::new();

    // Act
    let response = client
        // Use the returned application address
        .get(&format!("{}/lid/{}/bad-input", &ctx.address, format))
        .send()
        .await
        .expect("Failed to execute request.");

    // Assert
    assert_eq!(response.status(), StatusCode::BAD_REQUEST);
    // assert_eq!(Some(15), response.content_length());
    Ok(())
}
