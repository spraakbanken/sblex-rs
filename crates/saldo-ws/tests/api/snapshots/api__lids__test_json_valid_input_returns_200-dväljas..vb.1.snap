---
source: crates/saldo-ws/tests/api/lids.rs
expression: data
---
{
  "gf": "dväljas",
  "l": [
    "dväljas..1"
  ],
  "p": "vb_vs_dväljas"
}
